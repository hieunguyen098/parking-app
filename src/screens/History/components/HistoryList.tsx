import {
  ActivityIndicator,
  FlatList,
  RefreshControl,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  Text, View,
} from 'react-native';
import React, {useCallback, useState} from 'react';
import HistoryItem from './HistoryItem';
import {useFetchHistory} from "../../../hooks/useFetchHistory";
import {useSelector} from "react-redux";
import {HistoryType} from "../../../constants/History";
import {useFocusEffect} from "@react-navigation/native";

const HistoryList = ({type}: {type: string}) => {
  const user = useSelector((state: any) => state.auth.user);

  const [
    loading,
    histories,
    firstLoadHistory,
    loadOldHistory
  ] = useFetchHistory(type, user.phone)

  useFocusEffect(
      useCallback(() => {
        firstLoadHistory().then();
      }, [type])
  )

  return (
      <View style={styles.container}>
          <HistoriesList
              histories={histories}
              loadOldHistory={() => {}}
              firstLoadHistory={firstLoadHistory}
              loading={loading}
          />
      </View>
  );
};

const HistoriesList = ({histories, loadOldHistory, firstLoadHistory, loading}: any) => {
  return <FlatList
      data={histories}
      renderItem={({item, index}) => {
        return (
            <HistoryItem item={item}
                         style={index === histories.length - 1 ? {borderBottomWidth: 0} : {}}/>
        );
      }}
      keyExtractor={(item, index) => String(index)}
      onEndReached={loadOldHistory}
      onRefresh={firstLoadHistory}
      refreshing={loading}
      style={styles.contentContainer}
      ListEmptyComponent={<Text style={styles.emptyContainer}>Không có lịch sử giao dịch</Text>}
      ListFooterComponent={loading ? <ActivityIndicator size={1}/> : null}
  />
}

export default HistoryList;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  contentContainer: {
    addingHorizontal: 16,
    marginBottom: 8
  },
  emptyContainer: {
    color: "gray",
    margin: 20,
    textAlign: "center",
    fontSize: 20
  }
});
