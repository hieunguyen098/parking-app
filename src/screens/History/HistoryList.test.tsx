import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import HistoryList from './components/HistoryList';

test('test histoy item', () => {
    const component = (
        <NavigationContainer>
            <HistoryList />
        </NavigationContainer>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
