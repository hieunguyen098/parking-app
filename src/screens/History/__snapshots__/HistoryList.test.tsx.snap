// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test histoy item 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "0",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "1",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
        Object {
          "id": "2",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "3",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
        Object {
          "id": "4",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "5",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
        Object {
          "id": "6",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "7",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
        Object {
          "id": "8",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "9",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
        Object {
          "id": "10",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "11",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
        Object {
          "id": "12",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "13",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
        Object {
          "id": "14",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-50.000đ",
          "time": "20:20 - 28/11/2022",
          "title": "Mua vé tháng",
          "type": "month-ticket",
        },
        Object {
          "id": "15",
          "place": "Nhà xe Trường Đại học Bách Khoa Tp.HCM",
          "price": "-2.000đ",
          "time": "17:05 - 27/11/2022",
          "title": "Thanh toán phí gửi xe",
          "type": "parking",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginBottom": 8,
        "paddingHorizontal": 16,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Mua vé tháng
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              20:20 - 28/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -50.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Thanh toán phí gửi xe
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              17:05 - 27/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -2.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Mua vé tháng
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              20:20 - 28/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -50.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Thanh toán phí gửi xe
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              17:05 - 27/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -2.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Mua vé tháng
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              20:20 - 28/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -50.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Thanh toán phí gửi xe
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              17:05 - 27/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -2.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Mua vé tháng
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              20:20 - 28/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -50.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Thanh toán phí gửi xe
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              17:05 - 27/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -2.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Mua vé tháng
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              20:20 - 28/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -50.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 100,
                "paddingBottom": 16,
                "paddingTop": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              Thanh toán phí gửi xe
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                }
              }
            >
              Nhà xe Trường Đại học Bách Khoa Tp.HCM
            </Text>
            <Text
              style={
                Object {
                  "color": "#B3B3B3",
                  "fontSize": 12,
                }
              }
            >
              17:05 - 27/11/2022
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "marginLeft": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                }
              }
            >
              -2.000đ
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
